src/wallet/websocket.ts

```diff
@@ -0,0 +1,147 @@
+/**
```

Comment:
The intent of this `src/wallet` folder is to aggregate everything related to wallet-service, right?

Why don't we name it `src/wallet-service` instead?

Reply:
Because it is meant to replace the old implementations and be the default implementation for the `wallet`

I think we should keep it as `wallet` and replace the `new` folder which is the old facade to something else, maybe `old` 😂 

But I wouldn't do this refactor in this PR, I suggest opening an issue to do that. What do you think, @luislhl?

Reply:
Looks good. I like how the `new` folder is actually the `old` facade.